frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/EnvironmentVariantsCard/EnvironmentVariantsTable/EnvironmentVariantsTable.tsx

```diff
@@ -0,0 +1,184 @@
+                            {calculateVariantWeight(weight)} %
+                        </TextCell>
+                    );
+                },
+                sortType: 'number',
+            },
+            {
+                Header: 'Type',
+                accessor: 'weightType',
+                Cell: TextCell,
+                sortType: 'alphanumeric',
+            },
+            {
+                Header: 'Actions',
+                id: 'Actions',
+                align: 'center',
+                Cell: ({
+                    row: { original },
+                }: {
+                    row: { original: IFeatureVariant };
+                }) => (
+                    <VariantsActionCell
+                        variant={original}
+                        projectId={projectId}
+                        editVariant={onEditVariant}
+                        deleteVariant={onDeleteVariant}
+                    />
+                ),
+                disableSortBy: true,
+            },
+        ],
+        [projectId, variants, environment]
+    );
+
+    const initialState = useMemo(
+        () => ({
+            sortBy: [{ id: 'name', desc: false }],
+        }),
+        []
+    );
+
+    const { data, getSearchText } = useSearch(columns, searchValue, variants);
+
+    const { headerGroups, rows, prepareRow, setHiddenColumns } = useTable(
+        {
+            columns: columns as any[],
+            data,
+            initialState,
+            sortTypes,
+            autoResetSortBy: false,
+            disableSortRemove: true,
+            disableMultiSort: true,
+        },
+        useSortBy,
+        useFlexLayout
+    );
+
+    useEffect(() => {
+        const hiddenColumns = [];
+        if (isLargeScreen) {
+            hiddenColumns.push('weightType');
+        }
+        if (isMediumScreen) {
+            hiddenColumns.push('payload', 'overrides');
+        }
+        setHiddenColumns(hiddenColumns);
+    }, [setHiddenColumns, isMediumScreen, isLargeScreen]);
+
+    return (
+        <StyledTableContainer>
+            <SearchHighlightProvider value={getSearchText(searchValue)}>
+                <VirtualizedTable
+                    rows={rows}
+                    headerGroups={headerGroups}
+                    prepareRow={prepareRow}
+                />
+            </SearchHighlightProvider>
+            <ConditionallyRender
+                condition={rows.length === 0}
+                show={
+                    <ConditionallyRender
+                        condition={searchValue?.length > 0}
+                        show={
+                            <TablePlaceholder>
+                                No variants found matching &ldquo;
+                                {searchValue}
+                                &rdquo;
+                            </TablePlaceholder>
+                        }
+                    />
+                }
+            />
+        </StyledTableContainer>
+    );
+};
```

Comment:
Mostly an updated duplicate of `frontend/src/component/feature/FeatureView/FeatureVariants/FeatureVariantsList/FeatureVariantsList.tsx`.